### PR TITLE
feat(hotreload): Support x:Bind to properties in hot reload scenarios

### PR DESCRIPTION
**Pull request by davidjohnoliver**
## PR Type

What kind of change does this PR introduce?

- Feature

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

closes https://github.com/unoplatform/private/issues/246

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]
🚀

**Review by carldebilly — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs [hunk 1]
```diff
@@ -725,12 +726,19 @@ private Binding BuildBindingExpression(object? instance, XamlMemberDefinition me
...
+			if (xBindNode != null)
 			{
-				throw new InvalidOperationException("Unable to find Binding or TemplateBinding node");
+				binding.Source = rootInstance;
+				// TODO: here we should be setting Mode to OneTime by default, and we should also respect x:DefaultBindMode values set 
+				// further up in the tree.
 			}
 
-			foreach (var bindingProperty in (bindingNode ?? templateBindingNode)!.Members)
+			if(bindingNode == null && templateBindingNode == null && xBindNode == null)
```
Comment (robloo) [inline src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs:736]: nit: missing space after `if`. Shouldn't some automated checks catch these things?

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs:736]: I don't think we run linting on the ci.

**Comment by davidjohnoliver** [comment]
@jeromelaban license check seems to be stalled :/

**Pull request merged**